presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java

```diff
@@ -250,10 +250,10 @@ protected void assertTableColumnNames(String tableName, String... columnNames)
         assertEquals(actual, expected);
     }
 
-    private static void assertExceptionMessage(Exception exception, @Language("RegExp") String regex)
+    private static void assertExceptionMessage(String sql, Exception exception, @Language("RegExp") String regex)
     {
         if (!exception.getMessage().matches(regex)) {
-            fail(format("Expected exception message '%s' to match '%s'", exception.getMessage(), regex));
+            fail(format("Expected query '%s' to fail with exception message '%s' matching '%s'", sql, exception.getMessage(), regex));
```

Comment:
What if the query contains new lines? This is going to look weird. I'd rephrase it as "Expected exception message '%s' to match '%s' for query: %s"